samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/AssessmentSettingsBean.java

```diff
@@ -1987,4 +1988,36 @@ private void resetExtendedTime() {
         this.transitoryExtendedTime = null;
         this.editingExtendedTime = false;
     }
+    
+    public List<FacesMessage> getErrorMessages() {
+        List<FacesMessage> list = new ArrayList<>();
+        FacesContext context = FacesContext.getCurrentInstance();
+        for (Iterator<FacesMessage> iter = context.getMessages() ; iter.hasNext(); ) {
+        	FacesMessage message = iter.next();
+        	if (FacesMessage.SEVERITY_WARN.equals(message.getSeverity())) {
+        		list.add(message);
+        	}
+        }
```

Comment:
You could simplify this pattern with a stream operation.

Reply:
Thanks, you're right. It's far more elegant with stream :)